cmd/ncproxy/ncproxy.go

```diff
@@ -42,13 +63,17 @@ func (s *grpcService) AddNIC(ctx context.Context, req *ncproxygrpc.AddNICRequest
 	if req.ContainerID == "" || req.EndpointName == "" || req.NicID == "" {
 		return nil, status.Errorf(codes.InvalidArgument, "received empty field in request: %+v", req)
 	}
-	if client, ok := containerIDToShim[req.ContainerID]; ok {
+	s.containerIDToComputeAgent.rw.RLock()
+	agent, ok := s.containerIDToComputeAgent.cache[req.ContainerID]
```

Comment:
Seems like all three of these lines could be moved to a function on `computeAgentCache`, and then we just call `agent, ok := s.containerIDToComputeAgent.get(req.ContainerID)`.

Reply:
Done :) also made a `put` function on `containerIDToComputeAgent`